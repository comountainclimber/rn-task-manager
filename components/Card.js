import React from "react";
import { StyleSheet, View, ViewPropTypes } from "react-native";
import PropTypes from "prop-types";

export default function Card({ children, additionalStyles }) {
  return <View style={[styles.card, additionalStyles]}>{children}</View>;
}

Card.defaultProps = {
  additionalStyles: {}
};

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ]).isRequired,
  additionalStyles: ViewPropTypes.style
};

export const styles = StyleSheet.create({
  card: {
    backgroundColor: "#FFFFFF",
    padding: 12,
    marginBottom: 12,
    borderRadius: 4,
    width: "100%",
    borderColor: "#F0F0F0",
    borderWidth: 1.5
  }
});
