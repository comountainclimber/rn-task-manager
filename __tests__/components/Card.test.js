import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";

import Card from "../../components/Card";

it("it renders without crashing", () => {
  const tree = renderer
    .create(
      <Card>
        <Text>Hello snapshot card test</Text>
      </Card>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
