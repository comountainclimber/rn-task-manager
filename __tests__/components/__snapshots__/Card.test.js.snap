// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#F0F0F0",
        "borderRadius": 4,
        "borderWidth": 1.5,
        "marginBottom": 12,
        "padding": 12,
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <Text>
    Hello snapshot card test
  </Text>
</View>
`;
